src/item.cpp

```diff
@@ -309,6 +309,12 @@ item& item::ammo_unset()
     return *this;
 }
 
+item& item::set_damage( double qty )
```

Comment:
This should return `bool` (is destroyed) too.
Otherwise the only way to check for "is the item destroyed after setting damage" is to manually compare it with the max damage value, which goes against the encapsulation.

Chaining is a minor gimmick, not worth dropping consistency for.